independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -146,6 +146,12 @@
             buildContext.putInternal(Key.INJECTION_POINTS.asString(), Collections.unmodifiableList(injectionPoints));
             buildContext.putInternal(Key.OBSERVERS.asString(), Collections.unmodifiableList(observers));
             buildContext.putInternal(Key.BEANS.asString(), Collections.unmodifiableList(beans));
+            List<ScopeInfo> allScopes = BuiltinScope.getAllBuiltInScopes();
```

Comment:
It's rather minor but I don't like this kind of contract. I think that `BuiltinScope.getAllBuiltInScopes()` should either return an immutable list... or even better, just create a new list and then `Collections.addAll(scopes, BuiltinScope.values())`, no need for extra method.

Reply:
Ah, you actually need `ScopeInfo`. In that case, I would try with a stream ;-).

Reply:
In fact I dont really 'need' `ScopeInfo`, the only use case we have is in the Spring bits (that I know of) and there you need `DotName`. But I found it more useful to provide ScopeInfo as that seems more resourceful than just name.
I can do it the stream way.